src/background/controllers/index.ts

```diff
@@ -49,19 +148,17 @@ const controllers: {
     const {payload} = message;
     const {address, msg} = payload;
     return new Promise(async (resolve, reject) => {
-      const queue = await app.exec('wallet', 'getTxQueue');
-
-      if (queue.length) {
-        return reject(new Error('user has unconfirmed tx.'));
+      if (pendingPopupRequest !== null) {
+        return reject(new Error("Another transaction is already pending confirmation."));
       }
 
       app.exec('analytics', 'track', {
         name: 'Bob3 Sign',
       });
 
-      const requestJson = await app.exec('wallet', 'createSignMessageRequest', msg, address);
+      await app.exec('wallet', 'createSignMessageRequest', msg, address);
 
-      await app.exec('wallet', 'addTxToQueue', requestJson);
+      await app.exec('wallet', 'addTxToQueue');
```

Comment:
(applies to all methods below)
Why don't we want to save the tx? the tx queue is used to render different screens that read the tx data. See `useTXQueue`, etc.